GetStream/stream-chat-android
# [4608] [v6] Navigate to Thread Messages via PNs

#  Please review https://github.com/GetStream/stream-chat-android/pull/4635 first as it is used by this PR

### 🎯 Goal

Partially addresses #4608 

### 🛠 Implementation details

- Implemented fetching thread parent messages inside the notification loading worker if applicable and possible
- Implemented fetching messages from DB in `LogicRegistry` and created a `ChatClient.getMessageUsingCache()` extension function which exposes it.
- Implemented the ability to load threads from PNs and the `scrollToMessage()` method inside `MessageListController`

### 🎨 UI Changes

Only behavioral changes (navigating to thread)

### 🧪 Testing

#### The following should be tested on both XML and Compose apps.

Test set A - Navigating via PNs:

Prerequisite: Log in on two separate devices with different users

1. From `Device A` send a PN to a channel both device users are members of
2. From `Device B` click on the PN
3. Test this both when the app is in the background and after it has been killed, offline and online
4. Type something in the Composer on `Device B` and send it

Expected: Should navigate to the correct channel, and the new message sent from `Device B` should be send in the correct mode (Normal mode in this case)

1. From `Device A` send a PN to a thread both device users are members of (User on `Device B` had to have previously posted in the thread)
2. From `Device B` click on the PN
3. Test this both when the app is in the background and after it has been killed, offline and online
4. Type something in the Composer on `Device B` and send it

Expected: Should navigate to the correct channel, and the new message sent from `Device B` should be send in the correct mode (Thread mode in this case)

Test set B - Navigating to a quoted message: 

1. Find a message that is a reply to a previous message
2. Click on the quoted message

Expected: Should navigate to the quoted message

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media2.giphy.com/media/d1yaB10EzYHtKjkELJ/giphy.gif?cid=ecf05e47k37bfpsl7koqiumechzuzge7i2wixlnce7vvv5ts&rid=giphy.gif&ct=g)
